### PR TITLE
Port from gtkdoc to gi-docgen

### PR DESCRIPTION
**Pull request by hughsie**
Fixes https://github.com/hughsie/libgusb/issues/79

### CODE REVIEW

File: docs/meson.build [hunk 1]
```diff
@@ -1,3 +1,45 @@
...
+      '--no-namespace-dir',
+      '--content-dir=@0@'.format(meson.current_source_dir()),
+      '@INPUT1@',
+    ],
+    depends: [
+      libgusb_girtarget[0],
+    ],
+    build_by_default: true,
+    install: true,
+    install_dir: join_paths(datadir, 'doc'),
```
Comment (nacho) [inline docs/meson.build:40]: for future work you might want to stop using join_paths

File: meson.build [hunk 2]
```diff
@@ -140,4 +141,6 @@ root_incdir = include_directories('.')
 
 subdir('gusb')
 subdir('tools')
+if get_option('docs')
 subdir('docs')
```
Comment (nacho) [inline meson.build:145]: normally I would have indented here

**Review by nacho — APPROVED** [review]

File: meson.build (hunk 2 above)
Comment (hughsie) [reply meson.build:145]: Agree, fixed, thanks.

File: docs/meson.build (hunk 1 above)
Comment (hughsie) [reply docs/meson.build:40]: yup, a known issue -- I really dislike the `foo / bar / baz` modern style.

**Pull request merged**